src/github.com/getlantern/flashlight/client/chained.go

```diff
@@ -124,7 +124,7 @@ func (s *ChainedServerInfo) Dialer(deviceID string) (*balancer.Dialer, error) {
 		Label:   label,
 		Trusted: s.Trusted,
 		DialFN: func(network, addr string) (net.Conn, error) {
-			conn, err := d.Dial(network, addr)
+			conn, err := d(network, addr)
```

Comment:
Primary change along with the switch to just return the function in chained/dialer.go